### ISSUE TITLE
[Bug]: Suggestion opens even if current user didn't type the trigger

### ISSUE THREAD

**Issue by bjgrosse** [ueberdosis/tiptap#4183]
### Which packages did you experience the bug in?

suggestion

### What Tiptap version are you using?

2.0.0

### What’s the bug you are facing?

In a collaborative editor, if you have a suggestion-based extension (like Mention), and your cursor is at the same location as another user's when that user types the trigger character, you'll get the suggestion UI even tho you didn't type the character:

![msedge_id56qX6sj3](https://github.com/ueberdosis/tiptap/assets/14151299/de11191f-21ce-42c4-8200-921accd22b2b)

Seems like one way to solve this is to only show the UI if the trigger character was just typed, as opposed to any time the cursor is positioned after the character. Is there any way to configure this kind of behavior currently?

### What browser are you using?

Chrome

### What did you expect to happen?

I wouldn't expect one user triggering a suggestion causing the suggestion UI to show for other users with cursors in the same location.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [x] Yes, I’m a sponsor. 💖

**Comment by bjgrosse** [comment]
(Pro subscriber, btw.. sorry I clicked the wrong link!)

**Comment by romansp** [comment]
@bdbch I'd like to work on this one.

Would it be okay to check for `isChangeOrigin(transaction) === true` early in [`apply` of suggestion plugin](https://github.com/ueberdosis/tiptap/blob/c5cbea624bac8955bf6a25eb10f0dc143063d1c5/packages/suggestion/src/suggestion.ts#L378), set `next.active = false` and return?

I'm not sure if importing `isChangeOrigin` from another package is appropriate though `extension-drag-handle` already does it here: https://github.com/ueberdosis/tiptap/blob/c5cbea624bac8955bf6a25eb10f0dc143063d1c5/packages/extension-drag-handle/src/drag-handle-plugin.ts#L200

**Comment by bdbch** [comment]
Hey. I created a PR for this issue on https://github.com/ueberdosis/tiptap/pull/7384 - it should not be an automatic fix but it exposes a new option `shouldShow` for the suggestion options where you get access to the transaction and can run a manual check whether the popup should be triggered or not.

That way you can use the `isChangeOrigin` utility to prevent the popup if it's not a user-created trigger character.

```js
suggestion: {
  shouldShow: ({ props }) => isChangeOrigin(props.transaction),
}
```

**Comment by romansp** [comment]
@bdbch such a clean solution! Thank you very much.

I believe you'd want to negate `isChangeOrigin` check though.

```ts
suggestion: {
  shouldShow: props => !isChangeOrigin(props.transaction),
}
```

**Comment by bdbch** [comment]
Yep - I just released this with 3.15.1, feel free to try it out. This should be fixed by that.

**Comment by romansp** [comment]
Sorry to comment on the resolved issue but simple `shouldShow: true/false` logic doesn't seem to be enough here. It seems to have helped a little but menu can still open unexpectedly when another user is editing the document.

What I see is that `shouldShow` will be called multiple times with various transactions (or with an appendedTransaction) after a single `/` keystroke. And after returning `false` for `isChangeOrigin` transactions to avoid displaying menu, it may be immediately followed by another transaction that won't match this check at all and returning `true` will still open suggestion popup. For example it could be transaction for awareness change from cursor extension.

Maybe having access to `prev.isActive` similar to `allow` will help to compute it correctly. I'll try to create a minimal repro to investigate this a bit closer.

**Comment by bdbch** [comment]
Will reopen for now so we can track the issue again.
